### ISSUE TITLE
domain is not working in user scim2

### ISSUE THREAD

**Issue by TwinsDestiny** [wso2/product-is#5917]
IS verson : is-km-5.7.0
I have user store named cmcc.com .
get `https://is.lab.hxdi.cn/scim2/Users?domain=cmcc.com`

response
```
{
	"totalResults": 3,
	"startIndex": 1,
	"itemsPerPage": 3,
	"schemas": [
		"urn:ietf:params:scim:api:messages:2.0:ListResponse"
	],
	"Resources": [{
			"meta": {
				"created": "2019-07-16T11:26:57Z",
				"lastModified": "2019-07-16T11:26:57Z",
				"resourceType": "User"
			},
			"roles": [{
				"type": "default",
				"value": "Internal/everyone"
			}],
			"name": {
				"familyName": "luandawei"
			},
			"id": "c0c56eb0-3c2f-4769-aecc-6d8bb597defa",
			"userType": "User",
			"userName": "CMCC.COM/luandawei"
		},
		{
			"meta": {
				"created": "2019-07-16T14:25:06Z",
				"lastModified": "2019-07-16T14:25:06Z",
				"resourceType": "User"
			},
			"roles": [{
				"type": "default",
				"value": "Internal/everyone"
			}],
			"name": {
				"familyName": "luandawei"
			},
			"id": "8ac627ee-f3fe-4291-9729-b9e3039435cc",
			"userType": "User",
			"userName": "luandawei"
		},
		{
			"meta": {
				"created": "2019-06-13T06:03:46Z",
				"lastModified": "2019-07-05T16:23:52Z"
			},
			"displayName": "admin",
			"roles": [{
				"type": "default",
				"value": "Application/cfportal_localhost,Application/IS,Application/APIM,Internal/everyone,admin,Application/API_PUBLISHER,Application/API_STORE,Application/EI,Internal/subscriber,Internal/creator,Application/admin_DefaultApplication_PRODUCTION,Application/cfportal,Internal/publisher"
			}],
			"name": {
				"familyName": "admin"
			},
			"groups": [{
					"display": "Application/cfportal_localhost",
					"value": "49f6f2c7-c4cb-4908-a47c-86e7e4f8c544"
				},
				{
					"display": "Application/IS",
					"value": "55961e92-68e8-4a8d-a08f-bfca15c09aa5"
				},
				{
					"display": "Application/APIM",
					"value": "ea72ce1f-a0cb-485b-9138-288f0d620557"
				},
				{
					"display": "Application/API_PUBLISHER",
					"value": "dadabc2f-6826-4147-92ff-0a764c642d3c"
				},
				{
					"display": "Application/API_STORE",
					"value": "d81585d6-cc3c-476a-a891-70a553d50dc7"
				},
				{
					"display": "Application/EI",
					"value": "b25f9d5e-5095-4fd9-b7eb-9f24bf6cc15f"
				},
				{
					"display": "Application/admin_DefaultApplication_PRODUCTION",
					"value": "790819fa-61ae-482f-a353-63b74ceff586"
				},
				{
					"display": "Application/cfportal",
					"value": "797e90d3-ae1b-47ee-9edd-e546ca3cde5e"
				}
			],
			"id": "ed4827f1-bf4c-4f3b-9db6-cfb54dbdcccb",
			"userName": "admin"
		}
	]
}
```

I try to get `https://is.lab.hxdi.cn/scim2/Users?filter=userName+eq+cmcc.com/luandawei`
response
```
{
    "totalResults": 1,
    "startIndex": 1,
    "itemsPerPage": 1,
    "schemas": [
        "urn:ietf:params:scim:api:messages:2.0:ListResponse"
    ],
    "Resources": [
        {
            "meta": {
                "created": "2019-07-16T11:26:57Z",
                "lastModified": "2019-07-16T11:26:57Z",
                "resourceType": "User"
            },
            "roles": [
                {
                    "type": "default",
                    "value": "Internal/everyone"
                }
            ],
            "name": {
                "familyName": "luandawei"
            },
            "id": "c0c56eb0-3c2f-4769-aecc-6d8bb597defa",
            "userType": "User",
            "userName": "CMCC.COM/luandawei"
        }
    ]
}
```

I use IS 5.8.0,it works correctly.

**Comment by TwinsDestiny** [comment]
fixed in #4603 with IS5.8.0,but IS5.7.0 not fixed.

**Comment by TwinsDestiny** [comment]
org.wso2.carbon.identity.scim2.common_1.2.7.jar ----- is-km-5.7.0
SCIMUserMnager.class
```
public List<Object> listUsersWithGET(Node rootNode, int startIndex, int count, String sortBy, String sortOrder, String domainName, Map<String, Boolean> requiredAttributes)
    throws CharonException, NotImplementedException
  {
    if ((sortBy != null) || (sortOrder != null))
      throw new NotImplementedException("Sorting is not supported");
    if (rootNode != null) {
      return filterUsers(rootNode, requiredAttributes, startIndex, count, sortBy, sortOrder, domainName);
    }
    return listUsers(requiredAttributes);
```

org.wso2.carbon.identity.scim2.common_1.2.45.jar ----- is-5.8.0
SCIMUserMnager.class
```
public List<Object> listUsersWithGET(Node rootNode, Integer startIndex, Integer count, String sortBy, String sortOrder, String domainName, Map<String, Boolean> requiredAttributes)
    throws CharonException, NotImplementedException
  {
    startIndex = Integer.valueOf(handleStartIndexEqualsNULL(startIndex));
    if ((sortBy != null) || (sortOrder != null))
      throw new NotImplementedException("Sorting is not supported");
    if ((count != null) && (count.intValue() == 0))
      return Collections.emptyList();
    if (rootNode != null) {
      return filterUsers(rootNode, requiredAttributes, startIndex.intValue(), count, sortBy, sortOrder, domainName);
    }
    return listUsers(requiredAttributes, startIndex.intValue(), count, sortBy, sortOrder, domainName);
  }
```
